packages/marko-web-omeda-identity-x/middleware/resync-customer-data.js

```diff
@@ -0,0 +1,61 @@
+const { getAsArray } = require('@parameter1/base-cms-object-path');
+const resyncCookie = require('../utils/resync-cookie');
+const findEncryptedId = require('../external-id/find-encrypted-customer-id');
+const {
+  getOmedaCustomerRecord,
+  getOmedaLinkedFields,
+  setOmedaDemographics,
+  setOmedaDeploymentTypes,
+} = require('../omeda-data');
+
+/**
+ * A middleware to resync the IdentityX user data with the Omeda customer data. After syncing data,
+ * sets a cookie to prevent resyncs for the configured timeframe.
+ */
+module.exports = ({
+  brandKey,
+  omedaGraphQLClientProp = '$omedaGraphQLClient',
+}) => async (req, res, next) => {
+  const { identityX, [omedaGraphQLClientProp]: omedaGraphQLClient } = req;
+  if (!omedaGraphQLClient) throw new Error(`Unable to load the Omeda GraphQL API from the request using prop ${omedaGraphQLClientProp}`);
+
+  // Load the user and find their external ID
+  const { user } = await req.identityX.loadActiveContext();
+  const encryptedCustomerId = findEncryptedId({ externalIds: getAsArray(user, 'externalIds'), brandKey });
```

Comment:
Add `if(!user || !user.id) return next();` Then remove the user from the if check on L27